code/src/com/sixsq/nuvla/server/resources/deployment_set/utils.clj

```diff
@@ -430,6 +430,16 @@
                                     :nuvla/authn authn})]
     (:resources body)))
 
+(defn get-all-modules
+  [deployment-set applications-sets]
+  (->> (get-applications-sets deployment-set)
+       (mapcat merge-apps (module-utils/get-applications-sets applications-sets))
+       (map #(-> (crud/retrieve {:params         {:uuid          (str (u/id->uuid (:id %)) "_" (:version %))
```

Comment:
Also my single comment is we are already retrieving all modules in another place. Is it possible to reuse to avoid retrieving it twice?